soccer/src/soccer/ros2_temp/debug_draw_interface.cpp

```diff
@@ -0,0 +1,60 @@
+#include "debug_draw_interface.hpp"
+
+#include <spdlog/spdlog.h>
+
+#include <rj_constants/topic_names.hpp>
+
+namespace ros2_temp {
+
+DebugDrawInterface::DebugDrawInterface(Context* context, rclcpp::Executor* executor)
+    : context_(context) {
+    node_ = std::make_shared<rclcpp::Node>("_debug_draw_interface");
+    debug_draw_sub_ = node_->create_subscription<rj_drawing_msgs::msg::DebugDraw>(
+        viz::topics::kDebugDrawPub, rclcpp::QoS(10),
+        [this](rj_drawing_msgs::msg::DebugDraw::SharedPtr debug_draw) {
+            spdlog::info("Got message");
+            latest_[debug_draw->layer] = debug_draw;
+        });
+
+    executor->add_node(node_);
+}
+
+void DebugDrawInterface::run() {
+    const auto& color_to_qt = [](const rj_drawing_msgs::msg::DrawColor& color) {
+        return QColor::fromRgb(color.r, color.g, color.b);
+    };
+
+    for (const auto& [layer, debug_draw] : latest_) {
+        spdlog::info("Got layer {}", layer);
+        for (const auto& shapes : debug_draw->shapes) {
+            context_->debug_drawer.draw_shape_set(rj_convert::convert_from_ros(shapes.shapes),
+                                                  color_to_qt(shapes.color),
+                                                  QString::fromStdString(layer));
+        }
+        for (const auto& segment : debug_draw->segments) {
+            context_->debug_drawer.draw_segment(rj_convert::convert_from_ros(segment.segment),
+                                                color_to_qt(segment.color),
+                                                QString::fromStdString(layer));
+        }
+        for (const auto& pose : debug_draw->poses) {
+            // TODO: Handle poses
```

Comment:
We don't currently have a way of explicitly drawing poses, but eventually we will probably draw them as coordinate transforms (i.e. display the axes).

Reply:
what about arrows?

Reply:
I guess that would work, or in general any kind of line segment+marker. I'll add in color so it's usable.